taar/recommenders/collaborative_recommender.py

```diff
@@ -72,14 +72,20 @@ def can_recommend(self, client_data, extra_data={}):
         return False
 
     def recommend(self, client_data, limit, extra_data={}):
+        """We only get meaningful recommendation if a client has at least an
+        addon installed."""
+        if len(client_data.get('installed_addons', [])) == 0:
```

Comment:
This should never be the case, since `can_recommend` takes care of checking that. Unless... are you making this change because of the new recommendation manager? If that's the case, instead of replicating the `can_recommend` logic in the `recommend` method, can't we simply call the former and use `[]` if it returns `False`?

Reply:
@crankycoder what about this one?